### PR TITLE
Feature/127 add todo dialog

### PR DESCRIPTION
**Pull request by s-bogelund**

### CODE REVIEW

**Review by mbnees — APPROVED** [review]

File: todo-angular/src/app/app.component.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { Component } from '@angular/core';
+import { TodoService } from './../app/todo/todo.service';
```
Comment (PRP-ITM) [inline todo-angular/src/app/app.component.ts]: Remove line

File: todo-angular/src/app/todo/add-new/add-new-dialog/add-new-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import {
+  MatDialog,
+  MatDialogRef,
+  MAT_DIALOG_DATA,
+} from '@angular/material/dialog';
+import { NewTodoForm } from '../add-new-form/add-new-form.component';
+import { TodoModule } from '../../todo.module';
+
+export interface TodoDialogData {
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/add-new/add-new-dialog/add-new-dialog.component.ts]: Remove

File: todo-angular/src/app/todo/add-new/add-new-dialog/add-new-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  styleUrls: ['./add-new-dialog.component.scss'],
+})
+export class AddNewDialogComponent implements OnInit {
+  title: string = 'Add New Todo';
+  constructor(public dialogRef: MatDialogRef<AddNewDialogComponent>) {}
+
+  @Output() newTodo: EventEmitter<any> = new EventEmitter<any>();
+
+  onAddTodo(newTodo: any): void {
+    console.log(this.title);
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/add-new/add-new-dialog/add-new-dialog.component.ts]: Remove log statements

File: todo-angular/src/app/todo/add-new/add-new-form/add-new-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  selector: 'add-new-form',
+  templateUrl: './add-new-form.component.html',
+  styleUrls: ['./add-new-form.component.scss'],
+})
+export class AddNewFormComponent implements OnInit {
+  constructor() {}
+
+  @Output() addTodo: EventEmitter<TodoModel> = new EventEmitter<TodoModel>();
+
+  todoName = new FormControl('');
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/add-new/add-new-form/add-new-form.component.ts]: `todoName = new FormControl<string>("")`

File: todo-angular/src/app/todo/add-new/add-new-form/add-new-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  onSubmit(): void {
+    const newTodo: TodoModel = {
+      todoListId: 1,
+      todoListName: this.todoName.value as string,
+      todoListDesc: this.todoDescription.value as string,
+      todoListDeleted: false,
+      todoTasks: null,
+    };
+
+    console.log(newTodo);
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/add-new/add-new-form/add-new-form.component.ts]: remove log

File: todo-angular/src/app/todo/add-new/add-new-form/add-new-form.component.ts (hunk 4 above)
Comment (PRP-ITM) [reply todo-angular/src/app/todo/add-new/add-new-form/add-new-form.component.ts]:
```
form = new FormGroup({
todoName: new FormControl<string>(""),
todoDescription: new FomControl<stirng>(""),
});
```

const {todoName, todoDescription} = this.form.value;

File: todo-angular/src/app/todo/add-new/add-new.component.html [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<div>
+  <button mat-raised-button (click)="openDialog()">Open Modal</button>
+  <add-new-dialog class="hidden"></add-new-dialog>
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/add-new/add-new.component.html]: remove this line

File: todo-angular/src/app/todo/add-new/add-new.component.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import { AddNewDialogComponent } from './add-new-dialog/add-new-dialog.component';
+
+import {
+  MatDialog,
+  MatDialogRef,
+  MAT_DIALOG_DATA,
+  _closeDialogVia,
+} from '@angular/material/dialog';
+
+export interface DialogData {
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/add-new/add-new.component.ts]: Remove this

File: todo-angular/src/app/todo/todo.service.ts [hunk 8]
```diff
@@ -18,6 +18,24 @@ export class TodoService {
       .pipe(catchError(this.handleError<TodoModel[]>('getTodos', [])));
   }
 
+  addTodo(newTodo: TodoModel): Observable<TodoModel> {
+    console.log('newTodo', newTodo);
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/todo.service.ts]: remove console

File: todo-angular/src/app/todo/todo.service.ts [hunk 9]
```diff
@@ -18,6 +18,24 @@ export class TodoService {
...
+
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/todo.service.ts]: const {todoListId, ...rest} = newTodo;

File: todo-angular/src/app/todo/add-new/add-new.component.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+export class AddNewComponent {
+  constructor(
+    public dialog: MatDialog,
+    private readonly todoService: TodoService
+  ) {}
+
+  onNewTodoAdded(todo: any): void {
+    this.todoService.addTodo(todo).subscribe({
+      next: (todo) => {
+        console.log(todo);
```
Comment (PRP-ITM) [inline todo-angular/src/app/todo/add-new/add-new.component.ts]: todoAdded.emit();

**Review by PRP-ITM — CHANGES_REQUESTED** [review]

**Review by PRP-ITM — APPROVED** [review]

**Pull request merged**